### DISCUSSION TITLE
adding variants to tailwind.config does not work

### DISCUSSION THREAD

**Discussion by fredbouin** [tailwindlabs/tailwindcss#3629] [Help] [answered]
I'm using tailwind (v1.8) which comes pre-installed with the statamic 3 cms. It seems that I can't get any variants to work. Any hover classes or pseudo class of any sort that I try to add to the tailwind config file has not effect.

I tested from a fresh install of statamic by simply adding hover to padding in my tailwind config file, it looks like this:

```
module.exports = {
    purge: {
      content: [
          './resources/**/*.antlers.html',
          './resources/**/*.blade.php',
          './content/**/*.md'
       ]
    },
    important: true,
    theme: {
      extend: {},
    },
    variants: {
      extend: {
       padding: ['hover', 'focus'],
      }
    },
    plugins: [],
  }
```
 I then added hover:p-4 to an element on the home page html file like this:
```
<div class="text-xs font-bold uppercase text-gray-600 space-x-3">
        <a class="hover:bg-gray-300 p-2 rounded" href="{{ cp_url }}">Control Panel</a>
        <a class="hover:p-4 p-2 rounded" href="https://statamic.dev" target="_blank" rel="noopener noreferrer">Documentation</a>
        <a class="hover:bg-gray-300 p-2 rounded" href="https://statamic.com/support" rel="noopener noreferrer">Get Support</a>
    </div>
```

The hover class does not get applied.

**Comment by adamwathan** [comment, marked as answer]
The `extend` feature doesn't exist in Tailwind v1.8 for variants, have to use a different API:

https://v1.tailwindcss.com/docs/configuring-variants#extending-default-variants

**Reply by fredbouin** [reply]
thanks didn't notice that one!
